Sources/Amplitude/ObjC/ObjCPlugin.swift

```diff
@@ -32,17 +52,44 @@ public class ObjCPlugin: NSObject {
         self.type = type
         self.setup = setup
         self.execute = execute
+        self.flush = nil
     }
 
     @objc(initWithType:execute:)
     public init(type: PluginType, execute: @escaping (ObjCBaseEvent) -> ObjCBaseEvent?) {
         self.type = type
         self.setup = nil
         self.execute = execute
+        self.flush = nil
+    }
+
+    @objc(initWithType:setup:execute:flush:)
+    public init(
+        type: PluginType,
+        setup: @escaping (ObjCAmplitude) -> Void,
+        execute: @escaping (ObjCBaseEvent) -> ObjCBaseEvent?,
+        flush: @escaping () -> Void
+    ) {
+        self.type = type
+        self.setup = setup
+        self.execute = execute
+        self.flush = flush
+    }
+
+    @objc(initWithType:execute:flush:)
+    public init(
+        type: PluginType,
+        execute: @escaping (ObjCBaseEvent) -> ObjCBaseEvent?,
+        flush: @escaping() -> Void)
+    {
+        self.type = type
+        self.setup = nil
+        self.execute = execute
+        self.flush = flush
     }
 }
 
-class ObjCPluginWrapper: Plugin {
+class ObjCPluginWrapper: Plugin, EventPlugin {
```

Comment:
Should we maybe have a separate wrapper for Event Plugin?

Reply:
It is an internal class - I think, single wrapper class is ok here,
Maybe `flush()` function should be moved to some other protocol (`DestinationPlugin`, ...) from `EventPlugin` protocol - it looks like the function has  no much sense for before/enrichment plugins, for example.

Reply:
Got it, since it's internal I'm ok with the single class. Thanks